src/Http/CurlClient.php

```diff
@@ -0,0 +1,207 @@
+<?php
+/**
+ * File contains the implementation of a curl Client for HTTP exchanges.
+ * PHP Version 7
+ *
+ * @category PSR7
+ * @package  Http
+ * @author   John Saigle <john.saigle@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ *
+ * @see https://www.php-fig.org/psr/psr-7/
+ */
+namespace LORIS\http;
+
+use \Psr\Http\Message\RequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+use \Zend\Diactoros\Response;
+
+/**
+ * This class provides an easy-to-use wrapper around the PHP cURL functions for
+ * use with the LORIS API.  It allows users to quickly login and create requests
+ * without knowing the internals of cURL.  Other benefits include:
+ *      - Submitting the API prefix with every request
+ *      - Automatically JSON-encoding POST bodies
+ *      - Including the auth token automatically.
+ * Users of the class will be able to sequence API calls quickly and write unit
+ * tests.
+ *
+ * PHP Version 7
+ *
+ * @category PSR7
+ * @package  Http
+ * @author   John Saigle <john.saigle@mcgill.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris/
+ *
+ * @see https://www.php-fig.org/psr/psr-7/
+ */
+
+class CurlClient
+{
+
+    /**
+     * Sends a PSR-7 request and returns a PSR-7 response.
+     *
+     * Every technically correct HTTP response MUST be returned as is,
+     * even if it represents an HTTP error response or a redirect instruction.
+     * The only special case is 1xx responses, which MUST be assembled in the
+     * HTTP client.
+     *
+     * The client MAY do modifications to the Request before sending it. Because
+     * PSR-7 objects are immutable, one cannot assume that the object passed to
+     * ClientInterface::sendRequest() will be the same object that is actually
+     * sent. For example the Request object that is returned by an exception MAY
+     * be a different object than the one passed to sendRequest, so comparison
+     * by reference (===) is not possible.
+     *
+     * {@link https://github.com/php-fig/fig-standards/blob/master/accepted/
+     *                          PSR-7-http-message-meta.md#why-value-objects}
+     *
+     * @param RequestInterface $request Request object. See link above.
+     *
+     * @return ResponseInterface
+     */
+    public function sendRequest(RequestInterface $request): ResponseInterface
```

Comment:
You seem to have copied this from PSR-18 without having included the "implements" line for the class.

Reply:
I don't think we can claim to implement the whole interface. We are missing the Exception handling part; for the least.

Reply:
But yes, `sendRequest` is aiming at being compliant with PSR-18